### PR TITLE
Current game selector, Internet archive source and apply all for xbox.com

### PR DESCRIPTION
**Pull request by aenrione**
Here is some features that I have added and find useful while using this wonderful tool:
## Current Game
Whenever I tried to look for a game or send it to my console, I always had to look for the titleID and copy it manually. Now, you can select a game from the Table, and then it will automatically fill the necessary forms

image
image

*Note: when you click in the titleID or databaseID, it also gets copied to your clipboard

## Internet Archive
Since I just modded my console, I found out that `xboxunity.net` is currently down. So instead of doing everything manually, I have added a new source from [Internet Archive Dump](https://archive.org/download/xboxunity-covers-fulldump_202311/xboxunity-covers-fulldump/4156004D/)

image

## Apply All for `Xbox.com`

Also, since I am lazy, I have added a button that *automatically* applies all assets from the xbox.com search result (with a max of 3 ss)

image

## 

This changes should improve the workflow of people using the tool (myself included)

### CODE REVIEW

File: AuroraAssetEditor/Classes/InternetArchiveAssetDownloader.cs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+				var downloader = new InternetArchiveDownloader();
+				var downloadedCover = downloader.DownloadCover(this);
+				if (downloadedCover != null)
+				{
+					_cover = downloadedCover;
+				}
+				else
+				{
+					return null;
+				}
```
Comment (Swizzy) [inline AuroraAssetEditor/Classes/InternetArchiveAssetDownloader.cs:171]:
```suggestion
				_cover = downloader.DownloadCover(this);
```

There really is no need to check for null here, you are assigning the same value to _cover anyway regardless if the downloaded result is null.

File: AuroraAssetEditor/Controls/FtpAssetsControl.xaml.cs
Comment (Swizzy) [inline AuroraAssetEditor/Controls/FtpAssetsControl.xaml.cs:1]: This file is very difficult to review due to the massive amounts of changes that don't appear to really to much at all, i assume there's alot of whitespace changes here such as tabs to spaces? is that correct?

File: AuroraAssetEditor/Controls/FtpAssetsControl.xaml
Comment (Swizzy) [inline AuroraAssetEditor/Controls/FtpAssetsControl.xaml:1]: This file is very difficult to review due to the massive amounts of changes that don't appear to really to much at all, i assume there's alot of whitespace changes here such as tabs to spaces? is that correct?

From a quick glance there appears to be nothing actually changed here?

File: AuroraAssetEditor/Controls/OnlineAssetsControl.xaml.cs [hunk 2]
```diff
@@ -12,19 +12,27 @@ namespace AuroraAssetEditor.Controls {
...
+	using Classes;
+	using System.Linq;
+	using Image = System.Drawing.Image;
 
     /// <summary>
     ///     Interaction logic for OnlineAssetsControl.xaml
     /// </summary>
     public partial class OnlineAssetsControl {
-        private readonly BackgroundControl _background;
+		private enum SelectedItem
```
Comment (Swizzy) [inline AuroraAssetEditor/Controls/OnlineAssetsControl.xaml.cs:29]: The name of this enum should be something that explains what it is, "SelectedItem" is a terrible name as it suggests it's a property, a better name would be "OnlineAssetSources" or something like that.

File: AuroraAssetEditor/Controls/OnlineAssetsControl.xaml.cs
Comment (Swizzy) [inline AuroraAssetEditor/Controls/OnlineAssetsControl.xaml.cs:1]: Could you try to equalize the indentation of your changes here to the original? it's kinda tricky to read it in Github due to the difference in indentation making it harder to follow the changes - some of this doesn't appear to have changed at all while being shown as removed and added back.

File: AuroraAssetEditor/MainWindow.xaml [hunk 3]
```diff
@@ -11,14 +12,17 @@
...
                 <RowDefinition Height="Auto" />
                 <RowDefinition Height="*" />
             </Grid.RowDefinitions>
-            <Menu Grid.Row="0"
-                  HorizontalAlignment="Stretch"
-                  VerticalAlignment="Top"
-                  Background="White">
+            <Menu
+                  VerticalAlignment="Center"
+                  Background="White" Grid.ColumnSpan="2">
```
Comment (Swizzy) [inline AuroraAssetEditor/MainWindow.xaml:25]:
```suggestion
            <Menu VerticalAlignment="Center"
                  Background="White"
                  Grid.ColumnSpan="2"
                  Grid.Row="0">
```
You should always specify the row even if it's implicitly 0.

File: AuroraAssetEditor/MainWindow.xaml [hunk 4]
```diff
@@ -47,12 +51,25 @@
                 <MenuItem x:Name="EditMenu"
                           Header="Edit"
                           SubmenuOpened="EditMenuOpened" />
+                <MenuItem x:Name="GameSelector"
+                  Visibility="Hidden"
+                  Margin="100,0,0,0">
```
Comment (Swizzy) [inline AuroraAssetEditor/MainWindow.xaml:56]: Why the high margin?

File: AuroraAssetEditor/Models/Game.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+		public bool IsGameSelected
+		{
+			get => _isGameSelected;
+			set { _isGameSelected = value; OnPropertyChanged(nameof(IsGameSelected)); }
+		}
+
+		public event PropertyChangedEventHandler PropertyChanged;
+
+		protected void OnPropertyChanged(string propertyName)
```
Comment (Swizzy) [inline AuroraAssetEditor/Models/Game.cs:43]:
```suggestion
		protected void OnPropertyChanged([System.Runtime.CompilerServices.CallerMemberName] string propertyName = null)
```

Doing this allows you to not need the nameof(...), it'll be automatically filled in by default: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callermembernameattribute?view=netframework-4.8

**Review by Swizzy — CHANGES_REQUESTED** [review]

**Comment by aenrione** [comment]
Thanks a lot for the feedback, this was my first time programming in C# and I think I also messed up the indentation with Visual Studio by using Neovim Unix whitespaces. Will try to improve the code!

File: AuroraAssetEditor/MainWindow.xaml (hunk 4 above)
Comment (aenrione) [reply AuroraAssetEditor/MainWindow.xaml:56]: It is so the current game appears more in the center

**Comment by aenrione** [comment]
@Swizzy I have made the suggestions but I still cant figure out the indentation part and why is different to yours, What is your current setup, do you use tabs/spaces, and what Line Endings

**Comment by Swizzy** [comment]
> @Swizzy I have made the suggestions but I still cant figure out the indentation part and why is different to yours, What is your current setup, do you use tabs/spaces, and what Line Endings

For future reference; don't force push - just make new commits, that makes it easier to follow what actually changed. Using force push should almost never really be used - there are exceptions when it's ok, but in general for a PR it's something you shouldn't do as it makes things even harder for reviewers.

I'll try to do a offline review of those files and see - the indentation i would've used was tabs, but it could've been translated to spaces (like it is now a days in VS, not sure if it was back then or not). Linebreaks i've used has always been Windows style.

**Review by Swizzy — APPROVED** [review]

**Comment by Swizzy** [comment]
Just so you know, i fixed up things in your fork using Notepad++, it was mainly spaces - i went through all files and fixed it all up to be all spaces tho, that made the changes actually readable on the web aswell since it could tell what actually changed vs. not.

**Pull request merged**